src/main/webapp/app/exercises/programming/manage/update/wizard-mode/programming-exercise-update-wizard-grading.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, Input } from '@angular/core';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
+import { IncludedInOverallScore } from 'app/entities/exercise.model';
+import { AssessmentType } from 'app/entities/assessment-type.model';
+import { SubmissionPolicyType } from 'app/entities/submission-policy.model';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+    selector: 'jhi-programming-exercise-update-wizard-grading',
+    templateUrl: './programming-exercise-update-wizard-grading.component.html',
+    styleUrls: ['./programming-exercise-update-wizard.component.scss'],
+})
+export class ProgrammingExerciseUpdateWizardGradingComponent {
+    readonly IncludedInOverallScore = IncludedInOverallScore;
+
+    private translationBasePath = 'artemisApp.programmingExercise.wizardMode.gradingLabels.';
+
+    @Input() isImport: boolean;
+    @Input() programmingExercise: ProgrammingExercise;
+
+    @Input() staticCodeAnalysisAllowed: boolean;
+    @Input() onStaticCodeAnalysisChanged: () => void;
+    @Input() maxPenaltyPattern: string;
+
+    faQuestionCircle = faQuestionCircle;
+
+    constructor(private translateService: TranslateService) {}
+
+    getGradingSummary() {
+        const summary = [];
+
+        if (!this.programmingExercise.maxPoints) {
+            return '';
+        }
+
+        const exerciseType = this.programmingExercise.includedInOverallScore === IncludedInOverallScore.INCLUDED_AS_BONUS ? 'bonusExercise' : 'normalExercise';
+        const assessmentType = this.programmingExercise.assessmentType === AssessmentType.AUTOMATIC ? 'assessmentAutomatic' : 'assessmentSemiautomatic';
+        const replacements = {
+            exerciseType: this.translateService.instant(this.translationBasePath + exerciseType),
+            maxPoints: this.programmingExercise.maxPoints.toString(),
+            bonusPoints: (this.programmingExercise.bonusPoints ?? 0).toString(),
+            assessmentType: this.translateService.instant(this.translationBasePath + assessmentType),
+            submissionLimit: this.programmingExercise.submissionPolicy?.submissionLimit,
+            exceedingPenalty: this.programmingExercise.submissionPolicy?.exceedingPenalty,
+            maxPenalty: ((this.programmingExercise.maxPoints * (this.programmingExercise.maxStaticCodeAnalysisPenalty ?? 100)) / 100).toString(),
+        };
+
+        summary.push(this.translateService.instant(this.translationBasePath + 'points'));
+
+        if (this.programmingExercise.includedInOverallScore === IncludedInOverallScore.NOT_INCLUDED) {
```

Comment:
General note: We should probably find a simpler way of parsing exercise settings and e.g. generate strings from it.